spec/defines/dns_logging_channel_spec.rb

```diff
@@ -154,6 +154,17 @@
         it { is_expected.not_to compile }
       end
 
+      describe 'with log_type = file and file_size and file_version not set' do
+        let(:params) do
+          {
+            'log_type'      => 'file',
+            'file_path'     => '/path/to/file.log',
+          }
+        end
+
+        it { is_expected.to compile.with_all_deps }
```

Comment:
I generally like to have more specific validation, something like
```suggestion
        it { is_expected.to compile.with_all_deps }
        it { is_expected.to contain_concat__fragment("named.conf-logging-channel-#{title}.dns").with_content(%r{file "/path/to/file.log";});
```